docs/apis/calcite.md

```diff
@@ -0,0 +1,235 @@
+CsvParser csvParser = dag.addOperator("CSVParser", CsvParser.class);
+dag.addStream("KafkaToCSV", kafkaInput.outputPort, csvParser.in);
+```
+
+Once we define DAG with KafkaInputOperator and CSVParser, it can parse data from Kafka topic. Upto this point, this is a regular Apex application without SQL. After this, we can register the output of CSVParser as a table using [StreamEndpoint](https://github.com/apache/apex-malhar/blob/master/sql/src/main/java/org/apache/apex/malhar/sql/table/StreamEndpoint.java) to run a SQL statement. This way we can develop applications in fusion style where the DAG is part SQL and part regular apex DAG.
+
+The following code will describe how we can define StreamEndpoint. 
+ 
+```java  
+SQLExecEnvironment sqlEnv = sqlEnv.getEnvironment();
+Map<String, Class> fieldMapping = ImmutableMap.<String, Class>of("RowTime", Date.class,
+                                                                 "id", Integer.class,
+                                                                 "Product", String.class,
+                                                                 "units", Integer.class);
+sqlEnv = sqlEnv.registerTable("FROMCSV", new StreamEndpoint(csvParser.out, fieldMapping));
+```
+
+To read existing data stream, we need to register it as a table with SQL execution environment with the name of the table and StreamEndpoint. StreamEndpoint can serve as input table or output table in SQL. For input table configuration we need to initialise StreamEndpoint with immediate upstream operator's output port and fieldMapping or POJO class for input tuple. For output table configuration, we need to initialise StreamEndpoint with immediate downstream operator's input port and fieldMapping or POJO class for output tuple. Once we register StreamEndpoint as a table with a name in SQL Execution Environment, it can be used as a table in SQL statement similar to other endpoints.
+
+When executeSQL API is called, specified SQL is converted to DAG as described in the previous section. Both examples read CSV data from Kafka. But in pure style SQL example "KafkaInputOperator+ CSVParser" in DAG is created based on KafkaEndpoint usage while in fusion style example, it was already defined as part of DAG and same is extended with other operators as shown in the image below. 
+
+![image alt text](images/image_2.jpg "DAG for Fusion Style SQL Application")
+
+
+For all Apex-Calcite integration examples, click [here](https://github.com/apache/apex-malhar/tree/master/demos/sql). 
+
+
+# Ongoing developments
+
+Apache Apex-Calcite integration provides support for basic queries and efforts are going on to extend support for aggregations, sorting and other features using Tumbling Window, Hoping Window and Session Window.
+Support for JSON, XML and JDBC endpoint will also be implemented. The goal of this integration is to make developing a streaming application using SQL easy so that SQL Developers don't have to write any java code at all.
```

Comment:
will also be implemented => are also planned

{This sentence is a really nice way to wrapup!}